platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/UiSettings.java

```diff
@@ -370,6 +376,10 @@ public int getCompassGravity() {
    */
   @UiThread
   public void setCompassMargins(int left, int top, int right, int bottom) {
+    compassMargins[0] = left;
```

Comment:
Same instructions (with different variables) are also present in `setLogoMargins` and `setAttributionMargins` methods.
What about extracting it into a method passing an array and `left`, `top`, `right` and `bottom` variables and remove the duplication?